x-pack/plugins/security_solution/server/usage/detections/rules/get_metrics.ts

```diff
@@ -0,0 +1,125 @@
+import { getRuleObjectCorrelations } from './transform_utils/get_rule_object_correlations';
+
+// eslint-disable-next-line no-restricted-imports
+import { legacyGetRuleActions } from '../../queries/legacy_get_rule_actions';
+
+export interface GetRuleMetricsOptions {
+  kibanaIndex: string;
+  signalsIndex: string;
+  esClient: ElasticsearchClient;
+  savedObjectsClient: SavedObjectsClientContract;
+  logger: Logger;
+}
+
+export const getRuleMetrics = async ({
+  kibanaIndex,
+  signalsIndex,
+  esClient,
+  savedObjectsClient,
+  logger,
+}: GetRuleMetricsOptions): Promise<RuleAdoption> => {
+  try {
+    // gets rule saved objects
+    const ruleResults = await getDetectionRules({
+      esClient,
+      kibanaIndex,
+      maxPerPage: MAX_PER_PAGE,
+      maxSize: MAX_RESULTS_WINDOW,
+      logger,
+    });
+
+    // early return if we don't have any detection rules then there is no need to query anything else
+    if (ruleResults.length === 0) {
+      return {
+        detection_rule_detail: [],
+        detection_rule_usage: getInitialRulesUsage(),
+      };
+    }
+
+    // gets the alerts data objects
+    const detectionAlertsRespPromise = getAlerts({
+      esClient,
+      signalsIndex: `${signalsIndex}*`,
+      maxPerPage: MAX_PER_PAGE,
+      maxSize: MAX_RESULTS_WINDOW,
+      logger,
+    });
+
+    // gets cases saved objects
+    const caseCommentsPromise = getCaseComments({
+      savedObjectsClient,
+      maxSize: MAX_PER_PAGE,
+      maxPerPage: MAX_RESULTS_WINDOW,
+      logger,
+    });
+
+    // gets the legacy rule actions to track legacy notifications.
+    const legacyRuleActionsPromise = legacyGetRuleActions({
+      savedObjectsClient,
+      maxSize: MAX_PER_PAGE,
+      maxPerPage: MAX_RESULTS_WINDOW,
+      logger,
+    });
+
+    const [detectionAlertsResp, caseComments, legacyRuleActions] = await Promise.all([
+      detectionAlertsRespPromise,
+      caseCommentsPromise,
+      legacyRuleActionsPromise,
+    ]);
+
+    // create in-memory maps for correlation
+    const legacyNotificationRuleIds = getRuleIdToEnabledMap(legacyRuleActions);
+    const casesRuleIds = getRuleIdToCasesMap(caseComments);
+    const alertsCounts = getAlertIdToCountMap(detectionAlertsResp);
+
+    // correlate the rule objects to the results
+    const rulesCorrelated = getRuleObjectCorrelations({
+      ruleResults,
+      legacyNotificationRuleIds,
+      casesRuleIds,
+      alertsCounts,
+    });
+
+    // Only bring back rule detail on elastic prepackaged detection rules
+    const elasticRuleObjects = rulesCorrelated.filter((hit) => hit.elastic_rule === true);
+
+    // calculate the rule usage
+    const rulesUsage = rulesCorrelated.reduce(
+      (usage, rule) => updateRuleUsage(rule, usage),
+      getInitialRulesUsage()
+    );
+
+    return {
+      detection_rule_detail: elasticRuleObjects,
+      detection_rule_usage: rulesUsage,
+    };
+  } catch (e) {
+    // ignore failure, usage will be zeroed
+    logger.error(
```

Comment:
Do you want to show this to the user? I'm not sure if the strategy is the same across Kibana, but I usually just `INFO` the error in telemetry.

Reply:
I can change these from `error` to `info` to not worry users 👍 . Sounds good.